### PR TITLE
Static sectioned data source

### PR DESCRIPTION
**Pull request by heiberg**
Hey Dan,

I don't know if this is something you want in the framework, but I had a need for it in one of my own projects, so here it is.
## Summary

This PR introduces the class `StaticSectionDatasource`. A concrete implementation of `DatasourceType`, which is similar to `StaticDatasource` but has its content items organised in sections rather than in a flat array.

Sections are represented by a new protocol, `SectionType`. A section is only required to vend its items, but in practice users can choose to decorate their `SectionType` models with a title, an icon or anything else required to render their custom headers or footers.
## Rationale

I know that the main focus is on database-driven sources, but sometimes data is sectioned and a poor fit for a database.

My own need for this was to drive a settings screen in an app. The supported settings are naturally static in nature, but when presented they are grouped into sections of related cells.

Implementing `UITableViewDataSource` by hand to build this screen seemed barbaric.
## Change Details
- The new class, `StaticSectionDatasource`, is implemented in the file `Sectioned.swift`. The name of this file was chosen to match `Segmented.swift`.
- `StaticSectionDatasource` conforms to `SequenceType` and `CollectionType` for its sections, similar to how `StaticDatasource` exposes its items through the same protocols.
- Any `SectionType` implementation automatically conforms to  `SequenceType` and `CollectionType` for its items through protocol extensions.
- A suite of unit tests are added for `StaticSectionDatasource` in `StaticSectionDatasourceTests.swift`. 
- The file containing the existing tests for `StaticDatasource` was renamed to `StaticDatasourceTests.swift` for symmetry.
- A support class (`EventSection`) was added to `Models.swift` for the unit tests. It models a section of `Event` objects with a section title.
- `README.md` was amended to mention `StaticSectionDatasource` next to `StaticDatasource`.
## Miscellaneous

I was going to provide an example project using `StaticSectionDatasource`, but I see the old examples have been removed. If you decide to include examples again in the future I'm happy to provide one.

### CODE REVIEW

File: TaylorSource.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -219,6 +234,18 @@
...
+			children = (
+				C44F8F4E1CF99A0E008A8478 /* NSFRCDatasource.swift */,
+				C44F8F4F1CF99A0E008A8478 /* NSFRCFactory.swift */,
+				C44F8F501CF99A0E008A8478 /* NSFRCIndexedUpdates.swift */,
+				C44F8F511CF99A0E008A8478 /* NSFRCUpdateHandler.swift */,
+			);
+			name = CoreData;
+			path = Sources/CoreData;
+			sourceTree = SOURCE_ROOT;
+		};
```
Comment (heiberg) [inline TaylorSource.xcodeproj/project.pbxproj:248]: These files were already present in the repo, but not included in the project. I just added them for completeness.

**Comment by danthorpe** [comment]
Hi @heiberg - sorry, been crazy busy this weekend. Only just getting a chance to see this now. Will look it over! Thanks again for all the contributions - I'm still working on Swift 2.2 stuff for TaylorSource...

**Comment by heiberg** [comment]
Oh, no worries! I assumed as much when I saw Cocoaheads Stockholm tweet about your talk there. You are always super responsive and your efforts are greatly appreciated! 👍

**Comment by danthorpe** [comment]
@heiberg - sorry this took a little while to merge! Been crazy busy at work and with Operations.

But thanks for the contributions!

**Pull request merged**